### DISCUSSION TITLE
install free ssl on postal

### DISCUSSION THREAD

**Discussion by kiven943** [postalserver/postal#2165] [Help with using Postal] [answered]
When I want to install sll it gives me the error

/opt/postal/app/public does not exist or is not a directory
How can I prove it or what is the solution please?

**Comment by zdeneksvarc** [comment, marked as answer]
It depends on which service you want to use TLS certificates for. Whether for HTTPS termination or StartTLS SMTP. You can find inspiration for both at https://github.com/zdeneksvarc/postal-tls

**Reply by kiven943** [reply]
I could not understand it because I am still a beginner, can you help me please and thank you

**Reply by zdeneksvarc** [reply]
Encrypted connections are related to the postal system in several vectors. It depends on which one you have in mind. Whether you mean an encrypted connection to the web interface or an encrypted connection to SMTP. If on SMTP, then whether StartTLS or SMTPS. Of course, all vectors can be based on the same certificate that is related to the FQDN.

If you want a ready-made solution, then look at the link above and install everything within Docker Compose.

**Comment by willpower232** [comment]
The directory is now inside the container, you need to obtain the certificate files and place them in the config directory which is placed into the container. You can then reference the files from the config file to set tls_enabled to true.
